doc/gen_docs.go

```diff
@@ -0,0 +1,170 @@
+
+	buf := new(bytes.Buffer)
+
+	var flagString string
+	flags := cmd.NonInheritedFlags()
+	flags.SetOutput(buf)
+	if flags.HasAvailableFlags() {
+		flags.PrintDefaults()
+		flagString = buf.String()
+		buf.Reset()
+	}
+
+	var flagSlice []string
+	flagSplitter := regexp.MustCompile("\n *-") // clips first '-' in flags starting the second flag
+	numberOfFlags := strings.Count(flagString, "--")
+	flagSlice = flagSplitter.Split(flagString, numberOfFlags)
+	for i, flag := range flagSlice {
+		if i > 0 {
+			flagSlice[i] = "-" + flag // add clipped '-' back to flags
+		}
+	}
+
+	var parentFlagString string
+	parentFlags := cmd.InheritedFlags()
+	parentFlags.SetOutput(buf)
+	if parentFlags.HasAvailableFlags() {
+		parentFlags.PrintDefaults()
+		parentFlagString = buf.String()
+		buf.Reset()
+	}
+
+	headerScale := 0
+	var parentLink string
+	if cmd.HasParent() {
+		parent := cmd.Parent()
+		pname := parent.CommandPath()
+		link := pname + ".md"
+		link = strings.Replace(link, " ", "_", -1)
+		parentLink = fmt.Sprintf("* [%s](%s)\t - %s\n", pname, linkHandler(link), parent.Short)
+
+		headerScale = 1
+		for parent.HasParent() {
+			headerScale += 1
+			parent = parent.Parent()
+		}
+	}
+
+	var childrenLinks []string
+	children := cmd.Commands()
+	sort.Sort(byName(children))
+
+	for _, child := range children {
+		var childLink string
+		if !child.IsAvailableCommand() || child.IsAdditionalHelpTopicCommand() {
+			continue
+		}
+		cname := name + " " + child.Name()
+		link := cname + ".md"
+		link = strings.Replace(link, " ", "_", -1)
+		childLink = fmt.Sprintf("* [%s](%s)\t - %s\n", cname, linkHandler(link), child.Short)
+		childrenLinks = append(childrenLinks, childLink)
+	}
+
+	var relatedLinks []string
+	relatedCmds := cmd.RelatedCommands()
+
+	for _, relCmd := range relatedCmds {
+		var relatedLink string
+		if !relCmd.IsAvailableCommand() || relCmd.IsAdditionalHelpTopicCommand() {
+			continue
+		}
+		rname := relCmd.CommandPath()
+		link := rname + ".md"
+		link = strings.Replace(link, " ", "_", -1)
+		relatedLink = fmt.Sprintf("* [%s](%s)\t - %s\n", rname, linkHandler(link), relCmd.Short)
+		relatedLinks = append(relatedLinks, relatedLink)
+	}
+
+	var commandLink string
+	link := name + ".md"
+	link = strings.Replace(link, " ", "_", -1)
+	commandLink = fmt.Sprintf("%s", linkHandler(link))
+
+	autoGenTag := "Auto generated by spf13/cobra on " + time.Now().Format("2-Jan-2006") + "\n"
+
+	return &CmdOutline{
+		Name:          name,
+		Short:         short,
+		Long:          long,
+		UseLine:       useLine,
+		Example:       example,
+		Flags:         flagString,
+		FlagSlice:     flagSlice,
+		ParentFlags:   parentFlagString,
+		ParentLink:    parentLink,
+		ChildrenLinks: childrenLinks,
+		RelatedLinks:  relatedLinks,
+		CommandLink:   commandLink,
+		HeaderScale:   headerScale,
+		AutoGenTag:    autoGenTag,
+	}
+}
+
+func GenDocsCustomTemplate(cmd *cobra.Command, w io.Writer, linkHandler func(string) string, template *template.Template) error {
```

Comment:
I think this needs a comment with explanation of what each argument is and what it's used for.